test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+
+        # set an operator
+        self.server.manager(MGR_CMD_SET, SERVER, {'operators': 'pbsoper@*'})
+
+        # List all jobs in maintenance mode as operator
+        self.server.expect(
+            NODE, {'maintenance_jobs': jid2 + "," + jid3}, runas='pbsoper')
+
+        # set maintenance_jobs as root
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            msg = "Cannot set attribute, read only" +\
+                  " or insufficient permission  maintenance_jobs"
+            self.assertTrue(msg in e.msg[0])
+
+        # Set maintenance_jobs as operator
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas='pbsoper')
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            msg = "Cannot set attribute, read only" +\
+                  " or insufficient permission  maintenance_jobs"
+            self.assertTrue(msg in e.msg[0])
+
+        # Set maintenance_jobs as user
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas=TEST_USER)
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            self.assertTrue("Unauthorized Request" in e.msg[0])
+
+    def test_list_jobs_2(self):
+        """
+        Test to list maintenance_jobs when no job is admin-suspended
+        """
+
+        # Submit a few jobs
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+
+        # verify that all are running
+        self.server.expect(JOB, {'job_state=R': 3, 'substate=42': 3})
+
+        # list maintenance_jobs. It should be empty
+        try:
+            self.server.expect(
+                NODE, {'maintenance_jobs': ""}, id='vn[0]', max_attempts=10)
+        except Exception, e:
+            self.assertFalse(
+                e.rv, "FAIL:Node vn[0] has maintenance_jobs set")
+            print ("PASS:Node vn[0] do not have maintenance_jobs set")
+
+        # Regular suspend a job
+        self.server.sigjob(jid2, 'suspend', runas=ROOT_USER)
+
+        # List maintenance_jobs again
+        try:
+            self.server.expect(
+                NODE, {'maintenance_jobs': ""}, id='vn[0]', max_attempts=10)
+        except Exception, e:
+            self.assertFalse(
+                e.rv, "FAIL:Node vn[0] has maintenance_jobs set")
+            print ("PASS:Node vn[0] do not have maintenance_jobs set")
+
+    def test_preempt_order(self):
+        """
+        Test that sched_preempt_order has no impact
+        on admin-suspend
```

Comment:
I don't think this test accomplishes what you want.  Setting preempt_order without using preemption isn't going to accomplish much.